Composer/packages/client/src/recoilModel/dispatchers/project.ts

```diff
@@ -308,6 +309,25 @@ export const projectDispatcher = () => {
       );
       set(botProjectIdsState, [projectId]);
 
+      if (profile) {
+        // ABS Create Flow, update publishProfile after create project
+        const dispatcher = await snapshot.getPromise(dispatcherState);
+        const appId = profile.appId;
+        delete profile.appId;
+        const newProfile = {
+          name: `abs-${profile.botName}`,
+          type: 'azurePublish',
+          configuration: JSON.stringify({
+            hostname: profile.tag?.webapp || '',
+            runtimeIdentifier: 'win-x64',
+            settings: {
+              MicrosoftAppId: appId,
+            },
+            ...profile,
+          }),
+        };
+        dispatcher.setPublishTargets([newProfile], projectId);
+      }
```

Comment:
Can this be more generalized? What if several months from now, we do another creation flow integration and need to add a provisioning profile after creation that is not an azure publish profile?

If you look at the import flow I wrote for PVA, I tried to make the import flow itself generic, but abstracted away the PVA-specific logic into the `powerVirtualAgentsProvider.ts` 

Could a similar thing be done here where the publish profile type to be added is specified in the deep link payload, and then based on the publish profile type we fill out the rest of the publishing profile in another function / class?

I just worry about scale.